<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>图像视频处理器</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="QTabWidget" name="basicPage">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="imageTab">
       <attribute name="title">
        <string>图像处理</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1,0">
        <item>
         <widget class="QFrame" name="functionPanel">
          <property name="minimumSize">
           <size>
            <width>150</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>150</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::Shape::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Shadow::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QPushButton" name="applyAdjustmentsButton">
             <property name="toolTip">
              <string>将当前参数调整的结果保存为暂存区中的一个新副本</string>
             </property>
             <property name="text">
              <string>应用参数</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="Line" name="line">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="imageSharpenButton">
             <property name="text">
              <string>图像锐化</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="imageGrayscaleButton">
             <property name="text">
              <string>灰度化</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="cannyButton">
             <property name="text">
              <string>边缘检测</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="imageStitchButton">
             <property name="text">
              <string>图像组合</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="imageNewStitchButton">
             <property name="toolTip">
              <string>不同角度图片拼接成一张图片</string>
             </property>
             <property name="text">
              <string>图像拼接</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="imageBlendButton">
             <property name="text">
              <string>图像融合</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="textureMigrationButton">
             <property name="text">
              <string>纹理迁移</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="beautyButton">
             <property name="text">
              <string>一键美颜</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="gamma">
             <property name="text">
              <string>伽马变换</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="gammaSlider">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4" stretch="1,0">
          <item>
           <widget class="DroppableGraphicsView" name="graphicsView"/>
          </item>
          <item>
           <widget class="QFrame" name="bottomFrame">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>120</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>120</height>
             </size>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_2">
               <item>
                <widget class="QLabel" name="imageNameLabel">
                 <property name="text">
                  <string>图片名称：</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="imageResolutionLabel">
                 <property name="text">
                  <string>分辨率：</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="imageSizeLabel">
                 <property name="text">
                  <string>大小：</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QGridLayout" name="gridLayout">
               <item row="0" column="0">
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>亮度</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QSlider" name="brightnessSlider">
                 <property name="orientation">
                  <enum>Qt::Orientation::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item row="0" column="2">
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>饱和度</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="3">
                <widget class="QSlider" name="saturationSlider">
                 <property name="orientation">
                  <enum>Qt::Orientation::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>对比度</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QSlider" name="contrastSlider">
                 <property name="orientation">
                  <enum>Qt::Orientation::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item row="1" column="2">
                <widget class="QLabel" name="label_4">
                 <property name="text">
                  <string>色相</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="3">
                <widget class="QSlider" name="hueSlider">
                 <property name="orientation">
                  <enum>Qt::Orientation::Horizontal</enum>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="HistogramWidget" name="histogramWidget" native="true">
               <property name="minimumSize">
                <size>
                 <width>200</width>
                 <height>0</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_8">
               <item>
                <widget class="QLabel" name="colorSwatchLabel">
                 <property name="minimumSize">
                  <size>
                   <width>30</width>
                   <height>30</height>
                  </size>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_3">
                 <item>
                  <widget class="QLabel" name="colorRgbLabel">
                   <property name="text">
                    <string>RGB:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="colorHexLabel">
                   <property name="text">
                    <string>HEX:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="colorPosLabel">
                   <property name="text">
                    <string>Pos:</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QFrame" name="rencetImagePanel">
          <property name="minimumSize">
           <size>
            <width>150</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>150</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::Shape::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Shadow::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QListView" name="recentImageView"/>
           </item>
           <item>
            <widget class="QPushButton" name="deleteStagedImageButton">
             <property name="text">
              <string>删除选中</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="videoTab">
       <attribute name="title">
        <string>视频处理</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_10">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="1,4,1">
          <item>
           <widget class="QToolBox" name="videoEffectsToolBox">
            <property name="minimumSize">
             <size>
              <width>180</width>
              <height>0</height>
             </size>
            </property>
            <property name="currentIndex">
             <number>1</number>
            </property>
            <widget class="QWidget" name="page">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>196</width>
               <height>385</height>
              </rect>
             </property>
             <attribute name="label">
              <string>实时特效</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_7">
              <item>
               <widget class="QCheckBox" name="grayscaleCheckBox">
                <property name="text">
                 <string>灰度化</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="faceDetectCheckBox">
                <property name="text">
                 <string>人脸检测</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBox">
                <property name="title">
                 <string>滤波效果</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_6">
                 <item>
                  <widget class="QComboBox" name="filterComboBox"/>
                 </item>
                 <item>
                  <widget class="QSlider" name="filterIntensitySlider">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="page_2">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>196</width>
               <height>385</height>
              </rect>
             </property>
             <attribute name="label">
              <string>参数调试</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_11">
              <item>
               <layout class="QFormLayout" name="formLayout">
                <item row="0" column="0">
                 <widget class="QLabel" name="label_5">
                  <property name="text">
                   <string>亮度</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QSlider" name="videoBrightnessSlider">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_6">
                  <property name="text">
                   <string>对比度</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QSlider" name="videoContrastSlider">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="label_7">
                  <property name="text">
                   <string>饱和度</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QSlider" name="videoSaturationSlider">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                </item>
                <item row="3" column="0">
                 <widget class="QLabel" name="label_8">
                  <property name="text">
                   <string>色相</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="1">
                 <widget class="QSlider" name="videoHueSlider">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <spacer name="verticalSpacer_4">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>263</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="page_3">
             <attribute name="label">
              <string>导出操作</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <widget class="QPushButton" name="saveFrameButton">
                <property name="text">
                 <string>保存当前帧...</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="recordButton">
                <property name="text">
                 <string>开始录制</string>
                </property>
                <property name="checkable">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item>
           <widget class="DroppableGraphicsView" name="videoView"/>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_9">
            <item>
             <widget class="QListView" name="videoListView"/>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QPushButton" name="addVideoButton">
                <property name="text">
                 <string>添加</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="removeVideoButton">
                <property name="text">
                 <string>移除</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QWidget" name="controlBar" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QPushButton" name="playPauseButton">
             <property name="text">
              <string>播放</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="videoSlider">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="timeLabel">
             <property name="text">
              <string>00:00 / 00:00</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="speedComboBox"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>菜单</string>
    </property>
    <addaction name="actionopen"/>
    <addaction name="actionsave"/>
    <addaction name="actionsave_as"/>
    <addaction name="actionexit"/>
   </widget>
   <widget class="QMenu" name="edit">
    <property name="title">
     <string>编辑</string>
    </property>
    <addaction name="actionundo"/>
    <addaction name="actionredo"/>
   </widget>
   <widget class="QMenu" name="tool">
    <property name="title">
     <string>工具</string>
    </property>
   </widget>
   <widget class="QMenu" name="help">
    <property name="title">
     <string>帮助</string>
    </property>
   </widget>
   <addaction name="menu"/>
   <addaction name="edit"/>
   <addaction name="tool"/>
   <addaction name="help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionopen">
   <property name="text">
    <string>打开(O)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionsave">
   <property name="text">
    <string>保存(S)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionsave_as">
   <property name="text">
    <string>另存为(A)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionexit">
   <property name="text">
    <string>退出(X)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionundo">
   <property name="text">
    <string>撤销(Z)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionredo">
   <property name="text">
    <string>重做(R)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DroppableGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>droppablegraphicsview.h</header>
  </customwidget>
  <customwidget>
   <class>HistogramWidget</class>
   <extends>QWidget</extends>
   <header>histogramwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
